File: demo.in-jpaentity-med-validation/src/main/kotlin/com/example/demo/user_interface/constraints/UniqueKupeNummerConstraintValidator.java
package com.example.demo.user_interface.constraints;

import com.example.demo.user_interface.Kupe;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.util.List;

public class UniqueKupeNummerConstraintValidator implements ConstraintValidator<UniqueKupeNummer, List<Kupe>> {
    @Override
    public boolean isValid(List<Kupe> values, ConstraintValidatorContext context) {
        boolean duplicate = values.stream()
                .anyMatch(kupe1 ->
                        values.stream()
                                .anyMatch(kupe2 -> kupe1 != kupe2 && kupe1.getNummer() == kupe2.getNummer())
                );
        return !duplicate;
    }
}

File: in-jpaentity-with-validation/src/main/kotlin/com/example/in_jpaentity_with_validation/user_interface/constraints/UniqueKupeNummerConstraintValidator.kt
package com.example.in_jpaentity_with_validation.user_interface.constraints

import com.example.in_jpaentity_with_validation.user_interface.Kupe
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class UniqueKupeNummerConstraintValidator : ConstraintValidator<UniqueKupeNummer?, List<Kupe>> {
    override fun isValid(values: List<Kupe>, context: ConstraintValidatorContext): Boolean {
        val duplicate = values.stream()
                .anyMatch { kupe1: Kupe ->
                    values.stream()
                            .anyMatch { kupe2: Kupe -> kupe1 !== kupe2 && kupe1.nummer == kupe2.nummer }
                }
        return !duplicate
    }
}
